protocol/linkpreview/linkpreview_test.go

```diff
@@ -207,6 +207,54 @@ func Test_UnfurlURLs_YouTube(t *testing.T) {
 	assertContainsLongString(t, expected.Thumbnail.DataURI, preview.Thumbnail.DataURI, 100)
 }
 
+func Test_UnfurlURLs_giphy(t *testing.T) {
+	url := "https://www.giphy.com/stickers/happyplaceshow-transparent-sof2kXOSK5beJdb7xH"
+
+	expected := common.LinkPreview{
+		URL:      url,
+		Hostname: "www.giphy.com",
+		Title:    "Floating Tv Show Sticker by Happy Place for iOS & Android | GIPHY",
+		Thumbnail: common.LinkPreviewThumbnail{
+			URL: "https://media4.giphy.com/media/sof2kXOSK5beJdb7xH/giphy.gif",
+		},
+	}
+
+	transport := StubTransport{}
+	transport.AddURLMatcher(
+		"https://giphy.com/services/oembed",
+		[]byte(`
+		{
+			"title": "Floating Tv Show Sticker by Happy Place for iOS & Android | GIPHY",
+			"url": "https://media4.giphy.com/media/sof2kXOSK5beJdb7xH/giphy.gif",
+			"height": 480,
+			"width": 400,
+			"author_name": "Happy Place",
+			"author_url": "https://giphy.com/happyplaceshow",
+			"provider_name": "GIPHY",
+			"provider_url": "https://giphy.com/",
+			"type": "photo"
+		}
+		`),
+	)
+	transport.fallbackToDefaultTransport = true
```

Comment:
The idea is to not commit `fallbackToDefaultTransport = true` in the test. Otherwise what can happen is that if the developer set up an incorrect matcher in the transport (which is quite easy to happen), the test will make a real network call. This is useful during development, but we don't want that behavior in the CI because we want the tests to be fast and deterministic as much as possible.